packages/dev/core/src/Materials/Textures/renderTargetTexture.ts

```diff
@@ -58,12 +60,25 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
         return this._renderList;
     }
 
-    public set renderList(value: Nullable<Array<AbstractMesh>>) {
-        this._renderList = value;
+    private _renderListHasChangedObservable = new Observable<INotifyArrayChangeType<AbstractMesh>>();
+    private _renderListHasChangedObserver: Nullable<Observer<INotifyArrayChangeType<AbstractMesh>>> = null;
 
-        if (this._renderList) {
-            this._hookArray(this._renderList);
-        }
+    private _startObservingRenderListEvents() {
+        this._renderListHasChangedObserver = this._renderListHasChangedObservable.add(({ target, previousLength }) => {
+            if (target && previousLength && ((target.length > 0 && previousLength === 0) || (target.length === 0 && previousLength > 0))) {

```

Comment:
I think you want to check `previousLength !== undefined` and not simply `previousLength` in `target && previousLength`